libutils/json.c

```diff
@@ -376,7 +374,10 @@ int JsonCompare(const JsonElement *const a, const JsonElement *const b)
         return JsonContainerCompare(a, b);
 
     case JSON_ELEMENT_TYPE_PRIMITIVE:
-        return strcmp(a->primitive.value, b->primitive.value);
+        ret = strcmp(a->primitive.value, b->primitive.value);
+        if (ret)
+            Log(LOG_LEVEL_DEBUG, "JsonCompare() fails, primitive '%s' not equal to '%s'", a->primitive.value, b->primitive.value);
+        return ret;
```

Comment:
We normally declare the variable where it's used. In the case of switch statements, this normally issues a warning unless the variable is inside a new scope. Because it is potentially dangerous to have a conditionally declared variable in the outer scope.
```suggestion
        {
            int ret = strcmp(a->primitive.value, b->primitive.value);
            if (ret)
                Log(LOG_LEVEL_DEBUG, "JsonCompare() fails, primitive '%s' not equal to '%s'", a->primitive.value, b->primitive.value);
            return ret;
        }
```
If you accept this suggestion, also remove the declaration above :)

Reply:
Also, use explicit comparisons for integers (`if (ret != 0)`).